File: android/app/src/main/java/com/example/elcapi/BootReceiver.java
package com.example.elcapi;  // même package que MainActivity

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

// IMPORT DE MainActivity
import com.example.mqtt_hatab.MainActivity;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            Intent i = new Intent(context, MainActivity.class);
            i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            context.startActivity(i);
        }
    }
}

File: android/app/src/main/kotlin/com/example/mqtt_hatab/BootReceiver.kt
package com.example.mqtt_hatab

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            Log.d("BootReceiver", "Boot completed detected")

            // Par exemple, démarrer un service au boot
            val serviceIntent = Intent(context, BootService::class.java)
            context.startForegroundService(serviceIntent)
        }
    }
}
